### ISSUE TITLE
Error inside Docker image when building [docker build -t tiler .]  -> 'setuptools_scm.version.SetuptoolsOutdatedWarning: your setuptools is too old (<12)'

### ISSUE THREAD

**Issue by raspi-user** [Geovation/tiler#31]
Hi,
Ubuntu 16.04, any suggestions?

**sudo docker build -t tiler .**

```
Sending build context to Docker daemon 21.48 MB
Step 1/27 : FROM debian:jessie
 ---> fbc2ac014dc2
Step 2/27 : MAINTAINER James Milner<james.milner@geovation.uk>
 ---> Using cache
 ---> 1683f68694b0
Step 3/27 : RUN export DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 798d262e2ff5
Step 4/27 : ENV DEBIAN_FRONTEND noninteractive
 ---> Using cache
 ---> cd300c54300e
Step 5/27 : RUN dpkg-divert --local --rename --add /sbin/initctl
 ---> Using cache
 ---> b3214066a5ea
Step 6/27 : RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys B97B0AFCAA1A47F044F244A07FCC7D46ACCC4CF8
 ---> Using cache
 ---> 2cf3d82a66a6
Step 7/27 : RUN echo "deb http://apt.postgresql.org/pub/repos/apt/ jessie-pgdg main" > /etc/apt/sources.list.d/pgdg.list
 ---> Using cache
 ---> 5079ccae2fe9
Step 8/27 : RUN apt-get -y update
 ---> Using cache
 ---> 657b99c94815
Step 9/27 : RUN apt-get -y install ca-certificates rpl pwgen
 ---> Using cache
 ---> 25c0290f67e1
Step 10/27 : RUN apt-get install -y postgresql-9.5-postgis-2.2 netcat
 ---> Using cache
 ---> 83ef4e35f99a
Step 11/27 : ADD postgis/setup.sh /postgis/setup.sh
 ---> Using cache
 ---> a57c79a3149e
Step 12/27 : RUN chmod 0755 postgis/setup.sh
 ---> Using cache
 ---> 7f73a28b34ac
Step 13/27 : RUN postgis/setup.sh
 ---> Using cache
 ---> bbd5df6da16d
Step 14/27 : ADD postgis/start-postgis.sh /postgis/start-postgis.sh
 ---> Using cache
 ---> 306d203e19ab
Step 15/27 : RUN chmod 0755 /postgis/start-postgis.sh
 ---> Using cache
 ---> b068bbd21261
Step 16/27 : RUN apt-get install gdal-bin
 ---> Using cache
 ---> 4601676ae082
Step 17/27 : RUN apt-get install -y python-pip libpq-dev python-dev
 ---> Using cache
 ---> d43fcb393287
Step 18/27 : RUN pip install psycopg2 geojson nose coverage shapely mapbox-vector-tile
 ---> Running in e77e047301ee
Downloading/unpacking psycopg2
  Running setup.py (path:/tmp/pip-build-aF3_5m/psycopg2/setup.py) egg_info for package psycopg2
    
Downloading/unpacking geojson
  Downloading geojson-2.4.0-py2.py3-none-any.whl
Downloading/unpacking nose
Downloading/unpacking coverage
  Running setup.py (path:/tmp/pip-build-aF3_5m/coverage/setup.py) egg_info for package coverage
    /usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
      warnings.warn(msg)
    
    warning: no previously-included files found matching 'ci/appveyor.token'
    no previously-included directories found matching 'doc/_build'
    no previously-included directories found matching 'tests/eggsrc/build'
    no previously-included directories found matching 'tests/eggsrc/dist'
    no previously-included directories found matching 'tests/eggsrc/*.egg-info'
    warning: no previously-included files matching '*.py[co]' found anywhere in distribution
Downloading/unpacking shapely
  Running setup.py (path:/tmp/pip-build-aF3_5m/shapely/setup.py) egg_info for package shapely
    WARNING:/tmp/pip-build-aF3_5m/shapely/setup.py:Failed to determine system's GEOS version: Could not find geos-config script. Installation continuing. GEOS version will be checked on import of shapely.
    
Downloading/unpacking mapbox-vector-tile
  Downloading mapbox-vector-tile-1.2.0.tar.gz
  Running setup.py (path:/tmp/pip-build-aF3_5m/mapbox-vector-tile/setup.py) egg_info for package mapbox-vector-tile
    
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/lib/python2.7/dist-packages (from mapbox-vector-tile)
Downloading/unpacking protobuf (from mapbox-vector-tile)
Downloading/unpacking future (from mapbox-vector-tile)
  Running setup.py (path:/tmp/pip-build-aF3_5m/future/setup.py) egg_info for package future
    
    warning: no files found matching '*.au' under directory 'tests'
    warning: no files found matching '*.gif' under directory 'tests'
    warning: no files found matching '*.txt' under directory 'tests'
Downloading/unpacking pyclipper (from mapbox-vector-tile)
  Running setup.py (path:/tmp/pip-build-aF3_5m/pyclipper/setup.py) egg_info for package pyclipper
    zip_safe flag not set; analyzing archive contents...
    Distribution mode: Compiling Cython generated .cpp sources.
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "/tmp/pip-build-aF3_5m/pyclipper/setup.py", line 96, in <module>
        cmdclass=cmdclass,
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 262, in __init__
        self.fetch_build_eggs(attrs['setup_requires'])
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 287, in fetch_build_eggs
        replace_conflicting=True,
      File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 631, in resolve
        dist = best[req.key] = env.best_match(req, ws, installer)
      File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 874, in best_match
        return self.obtain(req, installer)
      File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 886, in obtain
        return installer(requirement)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 338, in fetch_build_egg
        return cmd.easy_install(req)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 636, in easy_install
        return self.install_item(spec, dist.location, tmpdir, deps)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 666, in install_item
        dists = self.install_eggs(spec, download, tmpdir)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 856, in install_eggs
        return self.build_and_install(setup_script, setup_base)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 1078, in build_and_install
        self.run_setup(setup_script, setup_base, args)
      File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 1063, in run_setup
        run_setup(setup_script, args)
      File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 68, in run_setup
        DirectorySandbox(setup_dir).run(runner)
      File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 114, in run
        return func()
      File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 67, in runner
        _execfile(setup_script, ns)
      File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 43, in _execfile
        exec(code, globals, locals)
      File "/tmp/easy_install-u5sZy7/setuptools_scm_git_archive-1.0/setup.py", line 34, in <module>
        def run(self):
      File "/usr/lib/python2.7/distutils/core.py", line 111, in setup
        _setup_distribution = dist = klass(attrs)
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 266, in __init__
        _Distribution.__init__(self,attrs)
      File "/usr/lib/python2.7/distutils/dist.py", line 287, in __init__
        self.finalize_options()
      File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 301, in finalize_options
        ep.load()(self, ep.name, value)
      File "build/bdist.linux-x86_64/egg/setuptools_scm/integration.py", line 10, in version_keyword
      File "build/bdist.linux-x86_64/egg/setuptools_scm/version.py", line 44, in _warn_if_setuptools_outdated
    setuptools_scm.version.SetuptoolsOutdatedWarning: your setuptools is too old (<12)
    Complete output from command python setup.py egg_info:
    zip_safe flag not set; analyzing archive contents...

Distribution mode: Compiling Cython generated .cpp sources.

Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/tmp/pip-build-aF3_5m/pyclipper/setup.py", line 96, in <module>

    cmdclass=cmdclass,

  File "/usr/lib/python2.7/distutils/core.py", line 111, in setup

    _setup_distribution = dist = klass(attrs)

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 262, in __init__

    self.fetch_build_eggs(attrs['setup_requires'])

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 287, in fetch_build_eggs

    replace_conflicting=True,

  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 631, in resolve

    dist = best[req.key] = env.best_match(req, ws, installer)

  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 874, in best_match

    return self.obtain(req, installer)

  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 886, in obtain

    return installer(requirement)

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 338, in fetch_build_egg

    return cmd.easy_install(req)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 636, in easy_install

    return self.install_item(spec, dist.location, tmpdir, deps)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 666, in install_item

    dists = self.install_eggs(spec, download, tmpdir)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 856, in install_eggs

    return self.build_and_install(setup_script, setup_base)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 1078, in build_and_install

    self.run_setup(setup_script, setup_base, args)

  File "/usr/lib/python2.7/dist-packages/setuptools/command/easy_install.py", line 1063, in run_setup

    run_setup(setup_script, args)

  File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 68, in run_setup

    DirectorySandbox(setup_dir).run(runner)

  File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 114, in run

    return func()

  File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 67, in runner

    _execfile(setup_script, ns)

  File "/usr/lib/python2.7/dist-packages/setuptools/sandbox.py", line 43, in _execfile

    exec(code, globals, locals)

  File "/tmp/easy_install-u5sZy7/setuptools_scm_git_archive-1.0/setup.py", line 34, in <module>

    def run(self):

  File "/usr/lib/python2.7/distutils/core.py", line 111, in setup

    _setup_distribution = dist = klass(attrs)

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 266, in __init__

    _Distribution.__init__(self,attrs)

  File "/usr/lib/python2.7/distutils/dist.py", line 287, in __init__

    self.finalize_options()

  File "/usr/lib/python2.7/dist-packages/setuptools/dist.py", line 301, in finalize_options

    ep.load()(self, ep.name, value)

  File "build/bdist.linux-x86_64/egg/setuptools_scm/integration.py", line 10, in version_keyword

  File "build/bdist.linux-x86_64/egg/setuptools_scm/version.py", line 44, in _warn_if_setuptools_outdated

setuptools_scm.version.SetuptoolsOutdatedWarning: your setuptools is too old (<12)

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /tmp/pip-build-aF3_5m/pyclipper
Storing debug log for failure in /root/.pip/pip.log
```

**Comment by wireman27** [comment]
Hey, I faced a similar issue. Here's what I did.

1. In the 'tiler' folder, open the file 'Dockerfile' with any text editor
2. Scroll down and look for this section:

RUN apt-get install -y python-pip libpq-dev python-dev
RUN pip install psycopg2 geojson nose coverage shapely mapbox-vector-tile

**change it to**

RUN apt-get install -y python-pip libpq-dev python-dev
RUN python -m pip install --upgrade pip setuptools wheel
RUN pip install psycopg2 geojson nose coverage shapely mapbox-vector-tile

Run the build command again.
